### PR TITLE
Fix RTL on Labels with FormattedText and/or LineHeight

### LINKED ISSUE #3311: [iOS] Incorrect Text Alignment for Label's FormattedText when FlowDirection = RightToLeft
### Steps to Reproduce
1. Create new project;
2. Add the following xaml to the main page ;

```
<StackLayout VerticalOptions="Center">
    <Label BackgroundColor="Yellow" FlowDirection="RightToLeft">
      <Label.FormattedText>
        <FormattedString>
          <Span Text="Formatted"></Span>
          <Span Text=" "></Span>
          <Span Text="Text"></Span>
        </FormattedString>
      </Label.FormattedText>
    </Label>
    <Label  Text="Normal Text" BackgroundColor="Pink" FlowDirection="RightToLeft"/>
  </StackLayout>
```
 
### Expected Behavior
The text should be aligned to the right 

### Actual Behavior
The text is aligned to the left.

### Basic Information

- Version with issue: (Latest) 3.1.0.637273
- IDE: Visual Studio 2017
- Platform Target Frameworks: 
  - iOS:  (Tested on) 11.4
- Affected Devices: All

### Screenshots

image

### LINKED ISSUE #8157: [Bug] [iOS] Label with HTML does not use applied text color
### Description

The HTML is always black on iOS even if TextColor is set to white.

### Steps to Reproduce

1. Place a Label with HTML content and a different text color

### Expected Behavior

Should display the text properly

### Actual Behavior

Always black text color

### Basic Information

- Version with issue: 4.3
- Last known good version: -
- IDE: Latest stable VS for Mac Enterprise
- Platform Target Frameworks: 
  - iOS: Tested with iOS 13.1

### Screenshots

The translator section is a Label with HTML content. It was previously correctly formatted with white text, centered and no misplaced characters. I was using the [HtmlLabel](https://github.com/matteobortolazzo/HtmlLabelPlugin) plugin previously. 
Simulator Screen Shot - iPhone 11 - 2019-10-22 at 21 39 30

### Reproduction Link

Way too easy to reproduce to make a whole reproduction repo for it...

### LINKED ISSUE #12273: Text Alignment Issue in IOS
I have observed another issue that the text alignment is not proper . I have two languages Arabic and English . in android it aligns them properly to right and left accordingly .
However in IOS there seems to be a problem, If I set flow direction , even English moves towards the right.

### LINKED ISSUE #12473: [Bug] iOS Label text alignment in RTL when mixing English/Arabic & LineHeight
### Description
Arabic text alignment randomly becomes incorrect for some items in the collection if an item in the collection has mixture of Arabic and English alphabets AND LineHeight attribute is set for the label.

I tried to narrow it down and this is my observation:
- If all items in the collection contain only Arabic alphabets, this issue doesn't appear. 
- If there are some items with mixture Arabic and English alphabets, but LineHeight is NOT set, also the items appear fine. 

I believe this issue is related to https://github.com/xamarin/Xamarin.Forms/pull/9795

### Basic Information
- IDE: VS Community 2019-16.7.4
- Platform Target Frameworks
  - iOS:  14.0
- Nuget Packages: XF: 4.8.0.1534 or 5.0.0.1558-pre3 , Essentials: 1.5.3.2 
- Affected Devices: iOS simulator

### Screenshots

ezgif com-gif-maker (1)

### Reproduction Link

Simple project is available in: https://github.com/darrabam/LabelAlignmentRTLBug

### LINKED ISSUE #12743: [Bug] UWP Hotreload not working for Font information changes on VS 2019 Preview
I know that UWP hotreload is in preview but i just wanted to give feedback. 
Basically it doesn't work for Font information changes like FontSize,  FontFamily etc. It just updates something irrelevant. 
When you restart the debugging session, you can see the change made by Hotreload is totally different than What actual update is after a restart.

 VS 2019  16.8.0 preview 6

### PR DESCRIPTION
**Pull request by rachelkang**
### Description of Change ###
Fix RTL so that it works on labels with FormattedText and/or LineHeight
Fix RTL so that it works on labels with other previously incompatible properties as well (i.e. FontAttributes, FontSize, LineHeight, TextColor, TextTransform, TextType, VerticalTextAlignment)

### Issues Resolved ### 
- fixes #3311 
- fixes #12473
- fixes #8157

### Platforms Affected ### 
- iOS

### Behavioral/Visual Changes ###
Labels with RTL (and with FormattedText and/or LineHeight) are properly right-aligned now

### Before/After Screenshots ### 

|   | **BEFORE** |**AFTER**   | 
|---|---|---|
|  Label w/ RTL and FormattedText | Simulator Screen Shot - iPod touch (7th generation) - 2021-01-15 at 14 57 36 | Simulator Screen Shot - iPod touch (7th generation) - 2021-01-15 at 14 56 26 |
| Label w/ RTL and LineHeight  |  Simulator Screen Shot - iPod touch (7th generation) - 2020-12-18 at 16 38 37 | Simulator Screen Shot - iPod touch (7th generation) - 2020-12-18 at 16 43 49  |

### Testing Procedure ###
- Run `Issue3311`
- Run `Issue12743`
- Make sure `RTLLabelWithFormattedTextWorks` and `RTLLabelWithLineHeightWorks` tests are passing

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by rmarinho** [comment]
Hey @rachelkang  the project isn't building, seems something wrong in the csproj

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems [hunk 1]
```diff
@@ -2052,6 +2057,8 @@
     <EmbeddedResource Include="$(MSBuildThisFileDirectory)Issue12084.xaml">
       <Generator>MSBuild:UpdateDesignTimeXaml</Generator>
     </EmbeddedResource>
+    <EmbeddedResource Include="$(MSBuildThisFileDirectory)Issue12473.xaml">
+      <SubType>Designer</SubType>
```
Comment (rmarinho) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems:2061]: you need to close the `EmbeddedResource` tag here

Comment (rachelkang) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems:2061]: Fixed! Thank you for catching that!!

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3311.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+						FormattedText = formattedString
+					}
+
+				}
+			};
+		}
+
+#if UITEST
+		[Test]
+		public void Issue1Test()
```
Comment (jsuarezruiz) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3311.cs:67]: Mnn, the name of the test should be "Issue3311Test", no?

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]
The UITest is failing:
Captura de pantalla 2021-01-14 a las 17 08 23
Could you take a look?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3311.cs (hunk 2 above)
Comment (rachelkang) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3311.cs:67]: yep, I just forgot to delete the template code. Will push up some changes in a bit!

**Review by Redth — APPROVED** [review]

**Review by jsuarezruiz — APPROVED** [review]

**Comment by mohsen-mohamed** [comment]
Arabic text is reversed "mirrored" if i used it as EmptyView for CollectionView 
this was an old bug and fixed some months ago bug it's now bugged again after this update

**Comment by hombyk-nevzi2-dAxzyd** [comment]
Xamarin.Forms.Platform.iOS/Renderers/LabelRenderer.cs

**Pull request merged**